Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,239 @@
+            var currentAgeText = currentAge.Text;
+
+            Assert.AreEqual(currentAgeText, ("39"));
+
+            var currentEmail = _driver.FindElement(By.XPath("(//*[@class=\"rt-tr-group\"]//div[@role='row'])[1]//div[@role='gridcell'][count(//*[contains(text(),'Email')]//ancestor::div[1]//preceding-sibling::div)]"));
+            var currentEmailText = currentEmail.Text;
+
+            Assert.AreEqual(currentEmailText, ("cierra@example.com"));
+
+            var currentSalary = _driver.FindElement(By.XPath("(//*[@class=\"rt-tr-group\"]//div[@role='row'])[1]//div[@role='gridcell'][count(//*[contains(text(),'Salary')]//ancestor::div[1]//preceding-sibling::div)]"));
+            var currentSalaryText = currentSalary.Text;
+
+            Assert.AreEqual(currentSalaryText, ("10000"));
+
+            var currentDepartment = _driver.FindElement(By.XPath("(//*[@class=\"rt-tr-group\"]//div[@role='row'])[1]//div[@role='gridcell'][count(//*[contains(text(),'Department')]//ancestor::div[1]//preceding-sibling::div)]"));
+            var currentDepartmentText = currentDepartment.Text;
+
+            Assert.AreEqual(currentDepartmentText, ("Insurance"));
+
+            var addButton = _driver.FindElement(By.XPath("//*[@id='addNewRecordButton']"));
+            addButton.Click();
+            
+            var firstNameTextBox = _driver.FindElement(By.XPath("//*[@id='firstName']"));
+            firstNameTextBox.SendKeys(firstName);
+            
+            var lastNameTextBox = _driver.FindElement(By.XPath("//*[@id='lastName']"));
+            lastNameTextBox.SendKeys(lastName);
+            
+            var userEmailTextBox = _driver.FindElement(By.XPath("//*[@id='userEmail']"));
+            userEmailTextBox.SendKeys(userEmail);
+            
+            var userAgeTextBox = _driver.FindElement(By.XPath("//*[@id='age']"));
+            userAgeTextBox.SendKeys(userAge);
+            
+            var userSalaryTextBox = _driver.FindElement(By.XPath("//*[@id='salary']"));
+            userSalaryTextBox.SendKeys(userSalary);
+            
+            var userDepartmentTextBox = _driver.FindElement(By.XPath("//*[@id='department']"));
+            userDepartmentTextBox.SendKeys(userDepartment);
+
+            var submitButton = _driver.FindElement(By.XPath("//*[@id='submit']"));
+            submitButton.Click();
+
+            var currentResult1 = _driver.FindElement(By.XPath("//*[@class='rt-tbody']/div[4]"));
+            var currentResultText1 = currentResult1.Text;
+
+            Assert.AreEqual(currentResultText1, ($"{firstName}\r\n{lastName}\r\n{userAge}\r\n{userEmail}\r\n{userSalary}\r\n{userDepartment}"));
+
+            var editButton = _driver.FindElement(By.XPath("//*[@id='edit-record-4']"));
+            editButton.Click();
+            
+            _driver.FindElement(By.XPath("//*[@id='salary']")).Clear();
+            _driver.FindElement(By.XPath("//*[@id='salary']")).SendKeys(editedUserSalary);
```

Comment:
do you really need to re-find the element second time?

Reply:
Fixed